Assets/Scripts/Game/Guilds/ThievesGuild.cs

```diff
@@ -76,6 +77,16 @@ public class ThievesGuild : Guild
 
         #region Guild Membership and Faction
 
+        public ThievesGuild()
+        {
+            RegisterEvents();
+        }
+
+        ~ThievesGuild()
+        {
+            UnregisterEvents();
```

Comment:
Same as for the DarkBrotherhood destructor: this is not needed (see my comment above).